File: test-case/tree/mirror/Mirror.scala
import soda.scala.unittest._

import scala.reflect.runtime.universe.typeOf

import soda.scala.leetcode._

import collection.mutable
import util.control.Breaks.{break, breakable}
object Solution {
  def mirror(root: TreeNode): TreeNode = {
    if (root == null) {
      return null
    }
    mirror(root.left)
    mirror(root.right)
    val temp = root.left
    root.left = root.right
    root.right = temp
    root
  }
}

class Solution {}

class Mirror extends (String => String) {
  override def apply(text: String): String = {
    val work = TestWork.forObject(typeOf[Solution], "mirror")
    // val work = TestWork.forStruct(...)
    // work.setValidator((R, R) => Boolean)
    work.compareSerial = true
    work.run(text)
  }
}

object Mirror extends App {
  println(new Mirror()(Utils.fromStdin()))
}

File: test-case/tree/mirror/Mirror.cs
using Soda.Unittest;
using Soda.Leetcode;

public class Solution {
    public TreeNode Mirror(TreeNode root)
    {
        if (root == null) {
            return null;
        }
        Mirror(root.left);
        Mirror(root.right);
        var temp = root.left;
        root.left = root.right;
        root.right = temp;
        return root;
    }
}

public class Mirror
{
    public static void Main(string[] args)
    {
        var work = WorkFactory.Create(Utils.Fn(new Solution().Mirror));
        // var work = WorkFactory.ForStruct<STRUCT>();
        // work.SetValidator((e, r) => ...);
        work.CompareSerial = true;
        Console.WriteLine(work.Run(Utils.ReadStdin()));
    }
}
